### PR TITLE
MULE-16554: Add new methods to check for key and mutability

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

**Comment by afelisatti** [comment]
—validate

**Comment by afelisatti** [comment]
--validate

File: src/main/java/org/mule/runtime/api/util/MultiMap.java [hunk 1]
```diff
@@ -39,10 +39,11 @@
  */
 public class MultiMap<K, V> implements Map<K, V>, Serializable {
 
-  private static final long serialVersionUID = -4723226524524565104L;
+  private static final long serialVersionUID = 5102672502150926191L;
 
   @SuppressWarnings("rawtypes")
   private static final MultiMap EMPTY_MAP = new MultiMap<>().toImmutableMultiMap();
+  private boolean mutable = true;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/util/MultiMap.java]: remove this attr and make this calculated

File: src/main/java/org/mule/runtime/api/util/MultiMap.java [hunk 2]
```diff
@@ -39,10 +39,11 @@
  */
 public class MultiMap<K, V> implements Map<K, V>, Serializable {
 
-  private static final long serialVersionUID = -4723226524524565104L;
+  private static final long serialVersionUID = 5102672502150926191L;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/util/MultiMap.java]: revert this

**Pull request merged**